### DISCUSSION TITLE
Max Number of DMX Ports Supported on ESP32 and Expansion Methods

### DISCUSSION THREAD

**Discussion by vjshaddow** [someweisguy/esp_dmx#154] [Q&A]
Hello 👋,

I'm working on an Art-Net/DMX node project and considering using the ESP32. I need to support multiple DMX ports (4-8). Can you please share insights on:

The maximum number of DMX ports the ESP32 can natively support?
Effective methods to expand beyond the native UART ports (using SPI/I2C expanders, etc.)?
Whether the ESP32 is up for this task, or should I consider a more capable IC like the STM32 (knowing I might be on my own for DMX and RDM libraries)?
Your guidance and suggestions would be greatly appreciated!

Thanks in advance!

**Comment by someweisguy** [comment]
Great question! The ESP32 supports a total of 3 UART ports; that is how many DMX ports it can handle. This library can use all three of them concurrently. 

I can't help much with expanding UART ports. This is not something I've researched. It is certainly possible to use 3 DMX ports on one ESP32 and also install an I2C or SPI driver on the ESP32. You could then use a different microcontroller to send I2C/SPI to multiple ESP32s and output multiple DMX universes that way. If you are looking for expansive RDM control, this may be a viable solution because as far as I know there are no other libraries which support RDM to the extent that this library does. 

Let me know if you have any other questions!
